### ISSUE TITLE
[ONNX Exporter Proposal] Support export of fixed BatchNorm layer in training mode

### ISSUE THREAD

**Issue by neginraoof** [pytorch/pytorch#52867]
## 🚀 Feature
Support export of fixed BatchNorm layer in training mode.
This use-case is needed for fine-tuning models.
Currently, ONNX exporter modifies the training mode of the batch_norm and dropout operators to match the training mode specified by export API.
However, this logic does not allow users to have certain fixed layers within the model and fine-tune other layers.

Current behavior:
```
Export API| Model    | Operator | Export Behavior  
----------------------------------------------------------------------------------
Training  | Eval     | Eval     | Model exported in *training* mode, with specific operator exported in *training*.
Training  | Eval     | Training | Model exported in *training* mode, with specific operator exported in *training*.
Training  | Training | Eval     | Model exported in *training* mode, with specific operator exported in *Training*.
Eval      | Training | Eval     | Model exported in *eval* mode, with specific operator exported in *eval*.
Eval      | Training | Training | Model exported in *eval* mode, with specific operator exported in *eval*.
Eval      | Eval     | Training | Model exported in *eval* mode, with specific operator exported in *eval*.

```
## Pitch

Exporter should not modify operators' training mode. We can still show a warning if the training mode of an operator does not match the model training mode.
Note that this is a breaking update to the exporter as it changes the previous behavior.

Expected behavior:

```
Export API| Model    | Operator | Export Behavior  
----------------------------------------------------------------------------------
Training  | Eval     | Eval     | Model exported in *training* mode, with specific operator exported in *eval*.
Training  | Eval     | Training | Model exported in *training* mode, with specific operator exported in *training*.
Training  | Training | Eval     | Model exported in *training* mode, with specific operator exported in *eval*.
Eval      | Training | Eval     | Model exported in *eval* mode, with specific operator exported in *eval*.
Eval      | Training | Training | Model exported in *eval* mode, with specific operator exported in *training*.
Eval      | Eval     | Training | Model exported in *eval* mode, with specific operator exported in *training*.
```
PR to fix this issue:
https://github.com/pytorch/pytorch/pull/52758

## Additional context
Will fix this issue:
https://github.com/pytorch/pytorch/issues/49229

cc @albanD @mruberry @jbschlosser @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof @SplitInfinity

**Comment by shinh** [comment]
I think this is equivalent to "Alternatives" in #49229. As I've written in #49229, I was not sure if a breaking change is allowed and I'd much prefer this fix. Thanks!

**Comment by neginraoof** [comment]
Will be resolved by: https://github.com/pytorch/pytorch/pull/52758
